core/schema/modulesource.go

```diff
@@ -1858,18 +1882,22 @@ func (s *moduleSourceSchema) runClientGenerator(
 		return genDirInst, fmt.Errorf("failed to load dependencies of this modules: %w", err)
 	}
 
-	// If the current module source has sources, we can transform it into a module
-	// to generate self bindings.
+	// If the current module source has sources and its SDK implements the `Runtime` interface,
+	// we can transform it into a moduleto generate self bindings.
 	if srcInst.Self.SDK != nil {
-		var mod dagql.Instance[*core.Module]
-		err = s.dag.Select(ctx, srcInst, &mod, dagql.Selector{
-			Field: "asModule",
-		})
-		if err != nil {
-			return genDirInst, fmt.Errorf("failed to transform module source into module: %w", err)
-		}
+		// We must make that if inside the condition to avoid checking a nil pointer on `SDKImpl`
+		// and make the engine crash.
```

Comment:
```suggestion
		// We must make sure that if inside the condition to avoid checking a nil pointer on `SDKImpl`
		// and make the engine crash.
```
The general wording of this could be clearer though.

Reply:
Done in [`8578e58` (#10525)](https://github.com/dagger/dagger/pull/10525/commits/8578e58c014beaba9e01f6ddade8c8fa8e18c9b2)